src/main.rs

```diff
@@ -897,21 +897,26 @@ async fn start_ldk() {
 		Arc::clone(&channel_manager),
 	));
 
-	// Start the CLI.
-	cli::poll_for_user_input(
-		Arc::clone(&peer_manager),
-		Arc::clone(&channel_manager),
-		Arc::clone(&keys_manager),
-		Arc::clone(&network_graph),
-		Arc::clone(&onion_messenger),
-		inbound_payments,
-		outbound_payments,
-		ldk_data_dir,
-		network,
-		Arc::clone(&logger),
-		Arc::clone(&persister),
-	)
-	.await;
+	// Exit if CLI polling exits or the background processor exits (which shouldn't happen unless we
+	// fail to write to the filesystem).
+	tokio::select! {
+		_ = cli::poll_for_user_input(
```

Comment:
Not sure if it is strictly required, but you may want to spawn this instead.

https://docs.rs/tokio/latest/tokio/macro.select.html#runtime-characteristics

Reply:
Based on those docs, I can't tell if with the way I had it written previously we would have gotten stuck on the CLI `select` branch. LMK if you know, for my understanding.

Reply:
Probably not stuck given the background processor task was already spawned and would have been running that future already.

https://docs.rs/tokio/latest/tokio/task/fn.spawn.html

Reply:
Ah, I meant whether we would ever hit the other `select` branch to check for the bg processor exiting. Looks like it does work: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0c2f28caa7c80e47d969465ba17e9986